### PR TITLE
feat: add batching query capability to mongodbindexer

### PR DESCRIPTION
**Pull request by davidbp**
This PR allows query to work on `keys:Iterable[str]`

### CODE REVIEW

File: indexers/keyvalue/MongoDBIndexer/__init__.py [hunk 1]
```diff
@@ -83,14 +83,16 @@ def query_handler(self) -> 'MongoDBHandler':
...
+    def query(self, keys: Iterable[str], *args, **kwargs) -> Optional[bytes]:
         """Query the serialized documents by document id.
 
         :param key: document id
         :return: serialized document
         """
+        result = []
         with self.query_handler as mongo_handler:
-            result = mongo_handler.query(key)
+            for key in keys:
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/__init__.py:94]: would be good to see if the mongo handler can handle itself multiple queries at the same time

Comment (davidbp) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: tried but did not work

Comment (JoanFM) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: see if can be changed to make it work

Comment (nan-wang) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]:
```
result = mongo_handler.query(keys)
```

Comment (nan-wang) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: change [mongodbhandler.py:L79](https://github.com/jina-ai/jina-hub/blob/master/indexers/keyvalue/MongoDBIndexer/mongodbhandler.py#L79) to 
```
cursor = self.collection.find({'_id': {"$in": keys}})
```

reference:

https://docs.mongodb.com/manual/tutorial/query-documents/#specify-conditions-using-query-operators

Comment (nan-wang) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: I think this should work.

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: indexers/keyvalue/MongoDBIndexer/__init__.py (hunk 1 above)
Comment (JoanFM) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: The problem may come from the fact that you cannot keep track of the `failing` keys. 

So maybe it is better to keep the single batch inside the `mongohandler`

Comment (davidbp) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:94]: This change that nan proposes works, the issue is that then I need to iterate over the returned list and verify the ids to check for missing ids;

I can do 
```aux = mongo_handler.query(['fccf9b08-acaf-11eb-bd09-0242ac110002', '221-fakeid-221',  'fccfda14-acaf-11eb-bd09-0242ac110002'])```

But then `len(aux)` is 2, not 3, and I need a loop over items.

File: indexers/keyvalue/MongoDBIndexer/tests/test_mongodbindexer.py [hunk 2]
```diff
@@ -73,5 +75,5 @@ def test_mongodbindexer():
 
     with MongoDBIndexer() as mongo_query:
         for key in keys:
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/tests/test_mongodbindexer.py:77]: can u do `assert mongo_query.query(keys) == [None, None...]`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: indexers/keyvalue/MongoDBIndexer/__init__.py [hunk 3]
```diff
@@ -83,14 +83,16 @@ def query_handler(self) -> 'MongoDBHandler':
...
+                result.append(mongo_handler.query(key))
 
         if result:
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/__init__.py:97]:
```suggestion
        return result
```

Comment (JoanFM) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:97]: no need for this check

File: indexers/keyvalue/MongoDBIndexer/tests/test_mongodbindexer.py (hunk 2 above)
Comment (JoanFM) [reply indexers/keyvalue/MongoDBIndexer/tests/test_mongodbindexer.py:77]: change the way you query in this test

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: indexers/keyvalue/MongoDBIndexer/__init__.py (hunk 3 above)
Comment (davidbp) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:97]: True,

**Review by JoanFM — APPROVED** [review]

**Pull request merged**